l10n_vn_account_invoice_number/models/account_invoice.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Trobz (http://trobz.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    tax_invoice_number = fields.Integer()
+    tax_invoice_form = fields.Char()
+    tax_invoice_series = fields.Char()
+    is_issuer_in_vn = fields.Boolean(string='Is issuer in Vietnam',
+                                     store=False,
+                                     compute='_compute_is_issuer_in_vn')
+
+    @api.onchange('partner_id', 'company_id')
+    def _onchange_partner_id(self):
+        self._compute_is_issuer_in_vn()
+        return super(AccountInvoice, self)._onchange_partner_id()
+
+    @api.multi
+    def _compute_is_issuer_in_vn(self):
+        if self.type == 'out_invoice' or self.type == 'in_refund':
```

Comment:
Cause `@api.multi`, maybe you have the `recordsets` ? You need `self.ensure_one()`

Reply:
Indeed, ideally it should be in a loop:
```
for rec in self:
    if rec.type == 'out_invoice' or rec.type == 'in_refund':
   ...
```

Reply:
Hi, could you please tell me, in which case it will need a loop?
so I could debug better.
thanks

Reply:
Almost every time you have a `@api.multi` because self might represent a recordset with multiple records. For instance if the field `is_issuer_in_vn` was displayed the tree view, self would contain many records in the function `_compute_is_issuer_in_vn`.